### ISSUE TITLE
Fine Tuning using the ESM model

### ISSUE THREAD

**Issue by wward97** [facebookresearch/esm#9]
Hi! 
Thanks for making this model public! I have had a good amount of success using it to predict function similar to how your example colab showed. I was wondering if you had any insight for how I could use the model as is to fine tune on more specific data? I think I could pretty reasonably set up a masking function by setting some values to 0 and using the logits and softmax to get predictions, but if I wanted to do next sequence predicting would I have to add a lot to the current model or is there an output I could use? Eg If a sequence is MYA i want to only use the information before the predicted AA and not the entire sequence. EG to predict M i would use only the fact that its after a start codon, then to predict Y i would use the previous hidden state that knows there is a start codon and M etc for the whole sequence.

**Comment by joshim5** [comment]
HI @wward97, glad to hear that you've found a good amount of success using ESM!

Am I understanding correctly that you'd like to use ESM to generate new sequences? One way is to perform several forward passes through the model. Concretely, to predict MYA, you could evaluate:
- `<bos><mask><mask><mask>` -> predict position `1`, corresponding to `M`
- `<bos>M<mask><mask>` -> predict position `2`, corresponding to `Y`
- `<bos>MY<mask>` -> predict position `3`, corresponding to `A`
In this setup, the model only has access to earlier amino acids in the sequence.

That said, while this setup would technically work, it's hard to say how strong the results would be, as this still an open research question. An alternative approach is to sample from the model using Gibbs Sampling. See [Wang & Cho 2019](https://arxiv.org/abs/1902.04094) for a discussion of sampling from masked language model.

By the way, you mentioned that you'd set up a masking function by setting values to zero. You'll probably getter better results if you replace the input with `alphabet.mask_idx`, because that's the masking token we used for pre-training.

**Comment by wward97** [comment]
Thank you that is very helpful! Good point ill use the prebuilt masking method. Thanks for the quick response and the help!
